signer/src/pubkey.rs

```diff
@@ -0,0 +1,152 @@
+//! Public key structures and algorithms
+//!
+//! Definition of public key structure and helpers
+
+use ark_ff::{BigInteger, PrimeField};
+use bs58;
+use core::fmt;
+use sha2::{Digest, Sha256};
+use std::ops::Neg;
+
+use crate::{BaseField, CurvePoint, FieldHelpers};
+
+/// Length of Mina addresses
+pub const MINA_ADDRESS_LEN: usize = 55;
+
+/// Public key
+#[derive(Copy, Clone, fmt::Debug, PartialEq, Eq)]
+pub struct PubKey(CurvePoint);
+
+impl PubKey {
+    /// Create a public key from curve point
+    pub fn new(point: CurvePoint) -> Self {
```

Comment:
is it important to do any validation check at this point? what if that pubkey is untrusted?

Reply:
also it feels like this function should be renamed `from` or something with `from_` as this is external input

Reply:
Good one!  Agreed -- thanks.